pushy/src/main/java/com/eatthepath/pushy/apns/util/LiveActivityApnsPayloadBuilder.java

```diff
@@ -0,0 +1,327 @@
+     *
+     * @return a reference to this payload builder
+     *
+     */
+    public LiveActivityApnsPayloadBuilder setAlertTitle(final String alertTitle) {
+        this.alertTitle = alertTitle;
+        return this;
+    }
+
+    /**
+     * <p>Sets the key of the title string in the receiving app's localized string list to be shown for the push
+     * notification. The message in the app's string list may optionally have placeholders, which will be populated by
+     * values from the given {@code alertArguments}.</p>
+     *
+     * @param localizedAlertTitleKey a key to a string in the receiving app's localized string list
+     * @param alertTitleArguments arguments to populate placeholders in the localized alert string; may be {@code null}
+     *
+     * @return a reference to this payload builder
+     *
+     * @see LiveActivityApnsPayloadBuilder#setAlertTitle(String)
+     */
+    public LiveActivityApnsPayloadBuilder setLocalizedAlertTitle(
+            final String localizedAlertTitleKey,
+            final String... alertTitleArguments
+    ) {
+        this.localizedAlertTitleKey = localizedAlertTitleKey;
+        this.localizedAlertTitleArguments = (alertTitleArguments != null && alertTitleArguments.length > 0) ? alertTitleArguments : null;
+
+        return this;
+    }
+
+    /**
+     * <p>Sets the name of the sound file to play when the push notification is received. According to Apple's
+     * documentation, the sound filename should be:</p>
+     *
+     * <blockquote>The name of a sound file in your app’s main bundle or in the {@code Library/Sounds} folder of your
+     * app’s container directory.</blockquote>
+     *
+     * <p>By default, no sound is included in the push notification.</p>
+     *
+     * @param soundFileName the name of the sound file to play, or {@code null} to send no sound
+     *
+     * @return a reference to this payload builder
+     *
+     * @see LiveActivityApnsPayloadBuilder#DEFAULT_SOUND_FILENAME
+     */
+    public LiveActivityApnsPayloadBuilder setAlertSound(final String soundFileName) {
+        this.soundFileName = soundFileName;
+        return this;
+    }
+
+    /**
+     * <p>Adds a custom property inside the content-state payload for Live Activities.</p>
+     *
+     * <p>The precise strategy for serializing the values of custom properties is defined by the specific
+     * {@code ApnsPayloadBuilder} implementation.</p>
+     *
+     * @param key the key of the custom property in the content-state object
+     * @param value the value of the custom property
+     *
+     * @return a reference to this payload builder
+     *
+     * @see <a href="https://developer.apple.com/documentation/activitykit/update-and-end-your-live-activity-with-remote-push-notifications">
+     *      Updating and ending your Live Activity with remote push notifications</a>
+     */
+    public LiveActivityApnsPayloadBuilder addContentStateProperty(final String key, final Object value) {
```

Comment:
Rather than specifying `content-state` on key at a time, why not just let callers provide a `Map<String, Object>`?

```suggestion
    public LiveActivityApnsPayloadBuilder setContentState(final Map<String, Object> contentState) {
```

I recognize that there's some precedent from `ApnsPayloadBuilder#addCustomProperty`, but we need to be working with individual keys there because we're asking callers to add to a map that has competing "owners;" we can't let callers clobber the `aps` property, for example. By contrast, callers "own" all of the `content-state` map and can just specify the whole thing in a single shot.